### DISCUSSION TITLE
Roadmap for Tarantool please.

### DISCUSSION THREAD

**Discussion by hiqsociety** [tarantool/tarantool#5836] [Q&A] [answered]
1. What can we expect with tarantool 3.0?

2. Advisable to use tarantool as primary data store to replace mysql entirely?

**Comment by hiqsociety** [comment]
Hi, what's the roadmap for 3.0?

**Comment by Mons** [comment]
Hello. Currently our roadmap is not public and is almost ready to became public.
Please, wait a little, I'll return with the answer.

**Comment by Mons** [comment, marked as answer]
In short, for nearest releases we have in plans:
* Stablilization and bugfixes in three most important recent features: Raft, Synchronous replication and new MVCC engine.
* Currently under development is an ability to create threaded modules (tarantool/xtm) and next-gen http server (tarantool/httpng) based on it
* At platform level for luajit we're creating memory profiler (take a look in 2.7.1) and interactive debugger. Also focusing on fixes of several LuaJIT pitfalls.
* Enhancements into type system between SQL, Lua and MsgPack
* For the maintenance and development side there are upcoming major changes in CLI
